tasks/release.py

```diff
@@ -117,3 +121,139 @@ def _find_v6_tag(ctx, v7_tag):
             print("Found v6 tag '{}'".format(v6_tag))
 
     return v6_tag
+
+
+def _is_version_higher(version_1, version_2):
+    if not version_2:
+        return True
+    if version_1["major"] < version_2["major"]:
+        return False
+    if version_1["minor"] < version_2["minor"]:
+        return False
+    if version_1["patch"] < version_2["patch"]:
+        return False
+    if version_1["rc"] < version_2["rc"]:
+        return False
+    return True
+
+
+def _create_version_dict_from_match(match):
+    groups = match.groups()
+    version = {
+        "major": int(groups[0]),
+        "minor": int(groups[1]),
+        "patch": int(groups[2]),
+        "rc": 0
+    }
+    if groups[4]:
+        version["rc"] = int(groups[4])
+    return version
+
+
+def _stringify_version(version_dict):
+    version = "{}.{}.{}" \
+        .format(version_dict["major"],
+                version_dict["minor"],
+                version_dict["patch"])
+    if version_dict["rc"] is not 0:
+        version = "{}-rc.{}".format(version, version_dict["rc"])
+    return version
+
+
+def _get_repo_rc_version(auth, repo, new_rc_version, version_re):
+    response = urllib.urlopen("{}api.github.com/repos/DataDog/{}/git/matching-refs/tags/{}.{}.{}"
+                              .format(auth,
+                                      repo,
+                                      new_rc_version["major"],
+                                      new_rc_version["minor"],
+                                      new_rc_version["patch"]))
+    tags = json.load(response)
+    highest_version = None
+    for tag in tags:
+        match = version_re.search(tag["ref"])
+        if match:
+            this_version = _create_version_dict_from_match(match)
+            if _is_version_higher(this_version, highest_version):
+                highest_version = this_version
+
+
+
+    version = "{}.{}.{}" \
+        .format(highest_version["major"],
+                highest_version["minor"],
+                highest_version["patch"])
+    if highest_version["rc"] is not 0:
+        version = "{}-rc.{}".format(version, highest_version["rc"])
+    return version
+
+
+@task
+def create_rc(ctx, omnibus_ruby_version = "datadog-5.5.0"):
```

Comment:
I think we could reuse most of this code to also create a `create_release` method which would do the same but for the final artifact. WDYT?

Reply:
Either that or two separate functions (might be useful to make sure you're meaning to create a new release entry and not an RC) which use common inner functions, as you see fit 👍